aws/resource_aws_apigatewayv2_authorizer.go

```diff
@@ -82,18 +94,42 @@ func resourceAwsApiGatewayV2Authorizer() *schema.Resource {
 func resourceAwsApiGatewayV2AuthorizerCreate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).apigatewayv2conn
 
+	apiId := d.Get("api_id").(string)
+	authorizerType := d.Get("authorizer_type").(string)
+
+	apiOutput, err := finder.ApiByID(conn, apiId)
+
+	if err != nil {
+		return fmt.Errorf("error reading API Gateway v2 API (%s): %s", apiId, err)
+	}
+
+	protocolType := aws.StringValue(apiOutput.ProtocolType)
+
 	req := &apigatewayv2.CreateAuthorizerInput{
-		ApiId:          aws.String(d.Get("api_id").(string)),
-		AuthorizerType: aws.String(d.Get("authorizer_type").(string)),
+		ApiId:          aws.String(apiId),
+		AuthorizerType: aws.String(authorizerType),
 		IdentitySource: expandStringSet(d.Get("identity_sources").(*schema.Set)),
 		Name:           aws.String(d.Get("name").(string)),
 	}
 	if v, ok := d.GetOk("authorizer_credentials_arn"); ok {
 		req.AuthorizerCredentialsArn = aws.String(v.(string))
 	}
+	if v, ok := d.GetOk("authorizer_payload_format_version"); ok {
+		req.AuthorizerPayloadFormatVersion = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("authorizer_result_ttl_in_seconds"); ok {
```

Comment:
I tested by passing 0 Explicitly on create and it sets 300. this should be probably `GetOkExists` for lack of a better option. same as in the V1 authorizer.

Reply:
Thanks for catching that slip.
I'll change and add another acceptance test that starts with an explicit value of `0` being set.

Reply:
Done.

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGatewayV2Authorizer_' ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSAPIGatewayV2Authorizer_ -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Authorizer_basic
=== PAUSE TestAccAWSAPIGatewayV2Authorizer_basic
=== RUN   TestAccAWSAPIGatewayV2Authorizer_disappears
=== PAUSE TestAccAWSAPIGatewayV2Authorizer_disappears
=== RUN   TestAccAWSAPIGatewayV2Authorizer_Credentials
=== PAUSE TestAccAWSAPIGatewayV2Authorizer_Credentials
=== RUN   TestAccAWSAPIGatewayV2Authorizer_JWT
=== PAUSE TestAccAWSAPIGatewayV2Authorizer_JWT
=== RUN   TestAccAWSAPIGatewayV2Authorizer_HttpApiLambdaRequestAuthorizer_InitialMissingCacheTTL
=== PAUSE TestAccAWSAPIGatewayV2Authorizer_HttpApiLambdaRequestAuthorizer_InitialMissingCacheTTL
=== RUN   TestAccAWSAPIGatewayV2Authorizer_HttpApiLambdaRequestAuthorizer_InitialZeroCacheTTL
=== PAUSE TestAccAWSAPIGatewayV2Authorizer_HttpApiLambdaRequestAuthorizer_InitialZeroCacheTTL
=== CONT  TestAccAWSAPIGatewayV2Authorizer_basic
=== CONT  TestAccAWSAPIGatewayV2Authorizer_HttpApiLambdaRequestAuthorizer_InitialMissingCacheTTL
--- PASS: TestAccAWSAPIGatewayV2Authorizer_basic (39.77s)
=== CONT  TestAccAWSAPIGatewayV2Authorizer_HttpApiLambdaRequestAuthorizer_InitialZeroCacheTTL
--- PASS: TestAccAWSAPIGatewayV2Authorizer_HttpApiLambdaRequestAuthorizer_InitialMissingCacheTTL (88.17s)
=== CONT  TestAccAWSAPIGatewayV2Authorizer_Credentials
--- PASS: TestAccAWSAPIGatewayV2Authorizer_HttpApiLambdaRequestAuthorizer_InitialZeroCacheTTL (73.00s)
=== CONT  TestAccAWSAPIGatewayV2Authorizer_JWT
--- PASS: TestAccAWSAPIGatewayV2Authorizer_Credentials (86.61s)
=== CONT  TestAccAWSAPIGatewayV2Authorizer_disappears
--- PASS: TestAccAWSAPIGatewayV2Authorizer_JWT (90.80s)
=== CONT  TestAccAWSAPIGatewayV2Authorizer_disappears
    resource_aws_apigatewayv2_authorizer_test.go:58: [INFO] Got non-empty plan, as expected
--- PASS: TestAccAWSAPIGatewayV2Authorizer_disappears (37.71s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	212.532s
```